lib/web/apiserver_test.go

```diff
@@ -472,6 +473,26 @@ func (s *WebSuite) createUser(c *C, user string, login string, pass string, otpS
 	}
 }
 
+func TestValidRedirectURL(t *testing.T) {
+	for _, tt := range []struct {
+		desc, url string
+		valid     bool
+	}{
+		{"valid absolute https url", "https://example.com?a=1", true},
+		{"valid absolute http url", "http://example.com?a=1", true},
+		{"valid relative url", "/path/to/something", true},
+		{"garbage", "fjoiewjwpods302j09", false},
+		{"empty string", "", false},
+		{"block bad protocol", "javascript:alert('xss')", false},
+	} {
+		tt := tt
+		t.Run(tt.desc, func(t *testing.T) {
+			t.Parallel()
```

Comment:
I would move `t.Parallel()` out of the subtests, and then you have no need for the `tt := tt` line.